vita3k/renderer/src/gl/compile_program.cpp

```diff
@@ -105,8 +105,210 @@ static SharedGLObject compile_spirv(GLenum type, const std::vector<std::uint32_t
     return shader;
 }
 
+static bool get_shaders_cache_hashs(GLState &renderer, const char *base_path, const char *title_id) {
+    const auto shaders_hashs_path{ fs::path(base_path) / "cache/shaders" / title_id / "hashs.dat" };
+    fs::ifstream shaders_hashs(shaders_hashs_path, std::ios::in | std::ios::binary);
+    if (shaders_hashs.is_open()) {
+        renderer.shaders_cache_hashs.clear();
+        // Read size of hashes list
+        size_t size;
+        shaders_hashs.read((char *)&size, sizeof(size));
+
+        // Check version of cache
+        size_t version_size;
+        shaders_hashs.read((char *)&version_size, sizeof(version_size));
+        std::vector<char> buffer(version_size);
+        shaders_hashs.read(buffer.data(), version_size);
+        const std::string versionInFile(buffer.begin(), buffer.end());
+        if (versionInFile != "1.00") {
+            LOG_WARN("Current version of cache: {}, is outdated, recreate it.", versionInFile);
+            return false;
+        }
+
+        // Read Hashe info value
+        for (size_t a = 0; a < size; a++) {
+            auto read = [&shaders_hashs]() {
+                size_t size;
+
+                shaders_hashs.read((char *)&size, sizeof(size));
```

Comment:
Should store it in binary instead of readable hex strimg, for smaller size!

Reply:
no idea how done this, this code is full based on app cache code writed by @1whatleytay

Reply:
Just dont convert it to hex string in write

Reply:
You can also ask taylor for more.

Reply:
> Just dont convert it to hex string in write

no understand your message here, is not hex string here, is original string give by game in fragment_program.hash, i convert it hex only for create/read shader file name

Reply:
Then cool. Why don't understand? Just yes or no.

Reply:
sorry,  i just said i no understrund you ask, you speek me no register hex string, but is no hex string, for that i no understand your message